pettingzoo/butterfly/pistonball/pistonball.py

```diff
@@ -630,30 +615,30 @@ def step(self, action):
 
         self.space.step(self.dt)
         if self._agent_selector.is_last():
-            ball_min_x = int(self.ball.position[0] - self.ball_radius)
-            ball_next_x = (
-                self.ball.position[0]
-                - self.ball_radius
-                + self.ball.velocity[0] * self.dt
+            ball_curr_pos = self._get_ball_position()
+            
+            # A rough, first-order prediction (i.e. velocity-only) of the balls next position.
+            # The physics environment may bounce the ball off the wall in the next time-step 
+            # without us first registering that win-condition. 
+            ball_predicted_next_pos = (
+                ball_curr_pos + 
+                self.ball.velocity[0] * self.dt
             )
-            if ball_next_x <= self.wall_width + 1:
+            # Include a single-pixel fudge-factor for the approximation.
+            if ball_predicted_next_pos <= self.wall_width + 1:
```

Comment:
Note this is my guess at the purpose of this logic! Please let me know if it looks wrong.

Reply:
Thanks for doing this! 

This looks good to me at first glance. Let me know when you're ready for a full review, and I'll tag 2 contributors that have (way) more experience with pistonball than me to have their feedback.

Keep up the good work and thank you for helping improve PettingZoo!

Reply:
Right on! :) 

I think it's ready for a full review now.